spec/td/command/job_spec.rb

```diff
@@ -13,32 +13,148 @@ module TreasureData::Command
+        it 'supports json output' do
+          file = Tempfile.new("job_spec")
+          command.send(:show_result, job, file, nil, 'json')
+          File.read(file.path).should == %Q([["1",2.0,{"key":3}],\n["4",5.0,{"key":6}],\n["7",8.0,{"key":9}]])
+        end
+
+        it 'supports csv output' do
+          file = Tempfile.new("job_spec")
+          command.send(:show_result, job, file, nil, 'csv')
+          File.read(file.path).should == %Q(1,2.0,"{""key"":3}"\n4,5.0,"{""key"":6}"\n7,8.0,"{""key"":9}"\n)
+        end
+
+        it 'supports tsv output' do
+          file = Tempfile.new("job_spec")
+          command.send(:show_result, job, file, nil, 'tsv')
+          File.read(file.path).should == %Q(1\t2.0\t{"key":3}\n4\t5.0\t{"key":6}\n7\t8.0\t{"key":9}\n)
         end
-        job
       end
 
-      it 'supports json output' do
-        file = Tempfile.new("job_spec")
-        command.send(:show_result, job, file, nil, 'json')
-        File.read(file.path).should == %Q([["1",2.0,{"key":3}],\n["4",5.0,{"key":6}],\n["7",8.0,{"key":9}]])
+      context 'result with nil' do
+        let :job do
+          job = TreasureData::Job.new(nil, 12345, 'hive', 'select * from employee')
+          job.instance_eval do
+            @result = [[[nil, 2.0, {key:3}], 1]]
+            @result_size = 3
+            @status = 'success'
+          end
+          job
+        end
+
+        context 'without --null option' do
+          it 'supports json output' do
+            file = Tempfile.new("job_spec")
+            command.send(:show_result, job, file, nil, 'json')
+            File.read(file.path).should == %Q([[null,2.0,{"key":3}]])
+          end
+
+          it 'supports csv output' do
+            file = Tempfile.new("job_spec")
+            command.send(:show_result, job, file, nil, 'csv')
+            File.read(file.path).should == %Q(null,2.0,"{""key"":3}"\n)
+          end
+
+          it 'supports tsv output' do
+            file = Tempfile.new("job_spec")
+            command.send(:show_result, job, file, nil, 'tsv')
+            File.read(file.path).should == %Q(null\t2.0\t{"key":3}\n)
+          end
+        end
+
+        context 'with --null option' do
+          it 'dose not effect json output (nil will be shown as null)' do
+            file = Tempfile.new("job_spec")
+            command.send(:show_result, job, file, nil, 'json', { null_expr: "NULL" })
+            File.read(file.path).should == %Q([[null,2.0,{"key":3}]])
+          end
+
+          it 'supports csv output' do
+            file = Tempfile.new("job_spec")
+            command.send(:show_result, job, file, nil, 'csv', { null_expr: "NULL" })
+            File.read(file.path).should == %Q(NULL,2.0,"{""key"":3}"\n)
+          end
+
+          it 'supports tsv output' do
+            file = Tempfile.new("job_spec")
+            command.send(:show_result, job, file, nil, 'tsv', { null_expr: "\"\"" })
+            File.read(file.path).should == %Q(""\t2.0\t{"key":3}\n)
```

Comment:
Would you please add a test case for checking '' (empty String)? I know it should work, simply because it's famous boundary value for test. :-)